Esri/arcgis-maps-sdk-swift-samples
# [Fix] Refactor `Augment reality to navigate route`

## Description

This PR refactors the `Augment reality to navigate route` sample to use a NavigationStack so that the navigation bar back button is present. This sample was refactored to match the setup of a similar AR sample `Augment reality to view hidden infrastructure`.

## Linked Issue(s)

- `swift/issues/5798`

## How To Test
- Navigate to the sample and select a route on the map
- Tap the camera button to view the AR scene view
- Verify that the back button works and the camera button can be opened again
